packages/clerk-js/src/core/events.ts

```diff
@@ -0,0 +1,25 @@
+import type { TokenResource } from '@clerk/types';
+
+export const eventBus = new EventTarget();
+
+export const createEvent = <T extends EventType>(eventName: string, data: T): CustomEvent => {
```

Comment:
This looks great!

I've got a few minor comments/ideas regarding the "ergonomics" of the internal event bus API but these can be either ignored or tackled in the future. 

1. Individual exports vs an `events` constant
Instead of exporting each event on its own, I'd probably group them under an `events` constant: 

```tsx
export const events = {
  TokenUpdate: 'token:update'
} as const;
```
(or even namespaced, eg `events.Token.Update`)

2. `createEvent` can constraint `eventName` based on the available event

3. We can create an association map between the event name and the event payload and use it in createEvent/parseEvent for extra type safety, a quick example: 
```
type TokenResource = {
    lastUpdatedAt: Date;
}

export const eventBus = new EventTarget();

export const events = {
    TokenUpdate: 'token:update',
    UserSignedIn: 'user:signedIn'
} as const

type Event = typeof events[keyof typeof events];
type EventType<T> = Record<string, unknown> & T;

type TokenUpdatePayload = { token: TokenResource | null };
type UserSignedInPayload = { user: { firstName: string } | null };

type EventPayload = {
    [events.TokenUpdate]: EventType<TokenUpdatePayload>,
    [events.UserSignedIn]: EventType<UserSignedInPayload>
}


export const createEvent = <E extends Event, P extends EventPayload[E]>(eventName: E, data: P): CustomEvent => {
    return new CustomEvent(eventName, { detail: data });
};

```
![image](https://user-images.githubusercontent.com/1811063/220581702-6ed2db61-4802-48ff-b18a-54202ea64bb6.png)

4. We can wrap eventBus and expose an `on` prop so we can have strict types on parseEvent as well and avoid the `addEventListener` etc boilerplate, eg: `eventBus.on(events.TokenUpdate, ({user}) => user....  )'. Similarly, we could have a `dispatch` method

5. (optional, nit-picky and debatable :) ) A final touch would be to make `.on` and `.dispatch` accept a function as the first argument pass the events as a prop so we don't need the explicit import, eg
```
eventBus.on(e => e.TokenUpdate, ({ user }) => {} )
```

Reply:
After discussing with @dimkl offline, we decided that point 5 makes no sense to let's ignore it :)